compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -117,9 +118,27 @@ MaterializeEncodingTypeConverter::MaterializeEncodingTypeConverter(
     if (failed(maybeEncodingInfo)) {
       return dropEncoding(type);
     }
-    return cast<RankedTensorType>(tensor::PackOp::inferPackedType(
+    auto encodingInfo = *maybeEncodingInfo;
+    auto packedType = cast<RankedTensorType>(tensor::PackOp::inferPackedType(
         tensorType, maybeEncodingInfo->innerTileSizes,
         maybeEncodingInfo->innerDimsPos, maybeEncodingInfo->outerDimsPerm));
+
+    if (!encodingInfo.swizzle) {
```

Comment:
Could add a short comment here that this is not used on CPU side, since CPU and GPU have ended up coupled in this function.